src/app/public/modules/phone-field/phone-field-input.directive.ts

```diff
@@ -0,0 +1,240 @@
+  NgControl,
+  FormControl
+} from '@angular/forms';
+
+import {
+  Subject
+} from 'rxjs/Subject';
+
+import {
+  SkyCountryData
+} from './types';
+
+import {
+  SkyPhoneFieldAdapterService
+} from './phone-field-adapter.service';
+
+import {
+  SkyPhoneFieldComponent
+} from './phone-field.component';
+
+// tslint:disable:no-forward-ref no-use-before-declare
+const SKY_PHONE_FIELD_VALUE_ACCESSOR = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => SkyPhoneFieldInputDirective),
+  multi: true
+};
+
+const SKY_PHONE_FIELD_VALIDATOR = {
+  provide: NG_VALIDATORS,
+  useExisting: forwardRef(() => SkyPhoneFieldInputDirective),
+  multi: true
+};
+// tslint:enable
+
+@Directive({
+  selector: '[skyPhoneFieldInput]',
+  providers: [
+    SKY_PHONE_FIELD_VALUE_ACCESSOR,
+    SKY_PHONE_FIELD_VALIDATOR,
+    SkyPhoneFieldAdapterService
+  ]
+})
+export class SkyPhoneFieldInputDirective implements OnInit, OnDestroy, AfterViewInit,
+  ControlValueAccessor, Validator {
+
+  @Input()
+  public set defaultCountry(value: string) {
+    this.skyPhoneFieldComponent.setDefaultCountry(value);
+    this._defaultCountryCode = value;
+  }
+
+  public get defaultCountry() {
+    return this._defaultCountryCode;
+  }
+
+  @Input()
+  public set disabled(value: boolean) {
+    this.skyPhoneFieldComponent.disabled = value;
+    this.adapterService.setElementDisabledState(this.elRef.nativeElement, value);
+    this._disabled = value;
+  }
+
+  public get disabled(): boolean {
+    return this._disabled || false;
+  }
+
+  @Input()
+  public formatModel: boolean = true;
+
+  @Input()
+  public skyPhoneFieldNoValidate: boolean = false;
+
+  @Output()
+  public selectedCountryChange = new EventEmitter<SkyCountryData>();
+
+  private set modelValue(value: string) {
+    this._modelValue = value;
+    this.adapterService.setElementValue(this.elRef.nativeElement, value);
+    if (value) {
+      let formattedValue = value;
+      if (this.formatModel) {
+        formattedValue = this.skyPhoneFieldComponent.formatNumber(value.toString());
+      }
+      this._onChange('+' + this.skyPhoneFieldComponent.selectedCountry.dialCode + ' ' + formattedValue);
+    } else {
+      this._onChange(value);
+    }
+    this._validatorChange();
+  }
+
+  private get modelValue(): string {
+    return this._modelValue;
+  }
+
+  private ngUnsubscribe = new Subject();
+
+  private _defaultCountryCode: string;
+
+  private _disabled: boolean;
+
+  private _modelValue: string;
+
+  public constructor(
+    private adapterService: SkyPhoneFieldAdapterService,
+    private changeDetector: ChangeDetectorRef,
+    private elRef: ElementRef,
+    private injector: Injector,
+    @Optional() public skyPhoneFieldComponent: SkyPhoneFieldComponent
+  ) { }
+
+  public ngOnInit(): void {
+    this.adapterService.addElementClass(this.elRef.nativeElement, 'sky-form-control');
+    if (this.defaultCountry) {
+      this.skyPhoneFieldComponent.selectCountry(this.defaultCountry);
```

Comment:
Since the parent component is `@Optional`, we should include a check at the beginning of the `ngOnInit` and throw an error if it doesn't exist.
See: https://github.com/blackbaud/skyux-datetime/blob/master/src/app/public/modules/datepicker/datepicker-input.directive.ts#L213-L218

Reply:
Oops. Good catch. Done.